HeterogeneousCore/SonicTriton/test/unittest.sh

```diff
@@ -16,6 +18,18 @@ if [ "$DEVICE" = "GPU" ]; then
 		echo "missing GPU"
 		exit 0
 	fi
+
+	SANDBOX=$(cmsTriton info)
+	# get sandbox env vars in subshell
+	CUDA_DRIVER_VERSION=$(source ${SANDBOX}/.singularity.d/env/10-docker2singularity.sh && echo $CUDA_DRIVER_VERSION)
+	# copied from https://github.com/triton-inference-server/server/blob/v2.11.0/nvidia_entrypoint.sh
+	DRIVER_VERSION=$(sed -n 's/^NVRM.*Kernel Module *\([0-9.]*\).*$/\1/p' /proc/driver/nvidia/version 2>/dev/null || true)
+	if [[ "${DRIVER_VERSION%%.*}" -ge "${CUDA_DRIVER_VERSION%%.*}" ]]; then
+		echo "has NVIDIA driver"
+	else
+		echo "missing (or too old) NVIDIA driver"
+		exit 0
```

Comment:
Rather than simply bailing out, you could try using the CUDA compatibility drivers that are shipped with CMSSW.

You can see how it is implemented for the CMSSW environment set up by SCRAM in https://github.com/cms-sw/cmssw-config/blob/scramv3/SCRAM/hooks/runtime/00-nvidia-drivers .

For more (somewhat confusing) information, see https://docs.nvidia.com/deploy/cuda-compatibility/ .

Reply:
Thanks, this is a definite improvement. The PR is updated with this now. I have a few questions and comments that I'll post in the main PR thread.